lib/multiplexer/multiplexer.go

```diff
@@ -75,6 +75,11 @@ const (
 // We define our own version to not create dependency on the 'services' package, which causes circular references
 type CertAuthorityGetter = func(ctx context.Context, id types.CertAuthID, loadKeys bool) (types.CertAuthority, error)
 
+type (
+	PreDetectFunc  = func(net.Conn) (PostDetectFunc, error)
+	PostDetectFunc = func(*Conn) net.Conn
```

Comment:
Do docs even apply to type aliases (as opposed to newtypes)?

Reply:
I added references to `Config` and `Mux`, since that's where the description of the assumed contract for both resides, and splitting it or duplicating it is IMO a bad idea.